sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -499,6 +507,14 @@ abstract class JdbcDialect extends Serializable with Logging {
   @Since("3.3.0")
   def isSupportedFunction(funcName: String): Boolean = false
 
+  /**
+   * Returns whether the database supports extract.
+   * @param extract The V2 Extract to be converted.
+   * @return True if the database supports extract.
+   */
+  @Since("4.1.0")
+  def isSupportedExtract(extract: Extract): Boolean = false
```

Comment:
I think we are making the API overly complicated. Too many knobs. It looks fine to ask the implementation to override `visitExtract` carefully.